src/contexts/_pubky.tsx

```
@@ -606,7 +611,15 @@ export function PubkyClientWrapper({ children }: { children: React.ReactNode })
       const postResult = specsBuilder!.createPost(content, kind, parentUri, embed, attachments);
 
       const post = postResult.post.toJson() as PubkyAppPost;
-      await homeserver.put(postResult.meta.url, JSON.stringify(post));
+
+      const response = await homeserver.put(postResult.meta.url, JSON.stringify(post));
+
+      if (!response.ok) {
+        if (response.status === 401) {
+          throw new Error('Session expired');
+        }
+        throw new Error(`HTTP error! status: ${response.status}`);
+      }
```

Creating a new Post is not the only thing that was failing silently. Also tagging, creating a follow, etc etc. This logic for Session expired needs to be somewhere else. Maybe inside of `homeserver.put()`